### PR TITLE
Recline support

### PR DESCRIPTION
**Pull request by anuveyatsu**

### CODE REVIEW

File: src/components/lineChart.js [hunk 1]
```diff
@@ -23,7 +23,13 @@ class Chart extends React.Component {
       .then((result) => {
 
         let dpJson = result.data
-        let dp = new DataPackage(dpJson)
+        let dp
```
Comment (rufuspollock) [inline src/components/lineChart.js:26]: Is dp a Data Package or a view or ...? If so I think this whole section might read better as:

```
// this should create DP object and do not much else
var dp = new DataPackage(dpJson, dataPackageUrl);
// we can then have methods like ...
var dataPackageView = dp.views[idx]
// which return a DataPackageView object with methods like ...
dataPackageView.raw // raw descriptor
dataPackageView.vegaSpec
dataPackageView.plotlySpec

// this is then used in our view factory (not sure this is the right pattern - but something like it)
var myVegaView = VizViewFactory.makeView(dataPackageView)

// PS: I think we want to rename view to viz in data package (view is overloaded)
```

Writing this I think we should chat about it.

File: src/components/lineChart.js [hunk 2]
```diff
@@ -39,6 +45,7 @@ class Chart extends React.Component {
   }
 
   componentDidUpdate() {
+    console.log(this.state.myVegaSpec)
```
Comment (rufuspollock) [inline src/components/lineChart.js]: Do we use console.log in production code? I assume not.

BTW: what is best logging framework for JS?

File: src/components/plotlyChart.js [hunk 3]
```diff
@@ -23,7 +23,13 @@ class PlotlyChart extends React.Component {
       .then((result) => {
 
         let dpJson = result.data
-        let dp = new ReclineView(dpJson)
+        let dp
```
Comment (rufuspollock) [inline src/components/plotlyChart.js:26]: This seems repetitive of lineChart - suggests we should be factoring out somewhere common.

**Review by rufuspollock — COMMENTED** [review]
Looks good generally - a nice simple commit to review.

One big point arising about how we design to support multiple chart rendering and multiple inputs that is worth thinking about. Let's chat about that before doing anything else (including merging).

**Comment by anuveyatsu** [comment]
I took into account all your comments @rufuspollock and already working on that. Just for now merging this PR as it is required to support recline view.

**Pull request merged**